--- iozone-3.140/src/current/iozone.c~	Sat Nov  9 02:30:48 2002
+++ iozone-3.140/src/current/iozone.c	Sat Nov  9 02:30:48 2002
@@ -221,6 +221,9 @@
   #define __USE_GNU
 #endif
 #include <fcntl.h>
+#if defined (linux) && defined (__powerpc__)
+#define O_DIRECT        0400000 /* direct disk access hint */
+#endif
 #if !defined(__FreeBSD__) && !defined(__OpenBSD__) && !defined(__APPLE__)
 #include <malloc.h>
 #endif
