diff -ur iozo/src/iozone.c ioz/src/iozone.c
--- iozo/src/iozone.c	Tue Sep 14 23:38:16 1999
+++ ioz/src/iozone.c	Tue Jul  4 00:52:42 2000
@@ -207,7 +207,7 @@
 typedef long long off64_t;
 #endif
 
-#ifndef solaris
+#if !defined(solaris) && !defined(linux)
 #ifndef _OFF64_T
 typedef long long off64_t;
 #endif
diff -ur iozo/src/libasync.c ioz/src/libasync.c
--- iozo/src/libasync.c	Tue Sep 14 23:38:17 1999
+++ ioz/src/libasync.c	Tue Jul  4 00:56:57 2000
@@ -95,7 +95,7 @@
  */
 
 #include <aio.h>
-#ifndef solaris
+#if !defined(solaris) && !defined(linux)
 #include <sys/timers.h>
 #endif
 #include <sys/errno.h>
diff -ur iozo/src/makefile ioz/src/makefile
--- iozo/src/makefile	Tue Sep 14 23:38:17 1999
+++ ioz/src/makefile	Tue Jul  4 01:06:09 2000
@@ -120,9 +120,10 @@
 #
 # GNU 'C' compiler Linux build with no threads, no largefiles, no async I/O 
 #
-linux: iozone_linux.o  libbif.o 
-	cc  -O -Dunix -DHAVE_ANSIC_C -DNO_THREADS -DSHARED_MEM \
-		-Dlinux iozone_linux.o libbif.o -o iozone
+linux: iozone_linux.o libbif.o libasync.o
+	gcc $(OPT) -Dunix -DHAVE_ANSIC_C -DASYNC_IO -D_LARGEFILE64_SOURCE \
+		-DSHARED_MEM -Dlinux -lpthread -lrt \
+		iozone_linux.o libbif.o libasync.o -o iozone
 
 # AIX
 # I would have built with ASYNC_IO but the AIX machine does not have 
@@ -351,10 +352,12 @@
 	@echo ""
 	@echo "Building iozone for Linux"
 	@echo ""
-	cc -c -O -Dunix -D_HPUX_SOURCE -DHAVE_ANSIC_C -DNO_THREADS \
-		-DSHARED_MEM -Dlinux iozone.c -o iozone_linux.o
-	cc -c -O -Dunix -D_HPUX_SOURCE -DHAVE_ANSIC_C -DNO_THREADS \
-		-DSHARED_MEM -Dlinux libbif.c -o libbif.o
+	gcc -c $(OPT) -Dunix -DHAVE_ANSIC_C -DASYNC_IO -D_LARGEFILE64_SOURCE \
+		-D_FILE_OFFSET_BITS=64 -DSHARED_MEM -Dlinux iozone.c -o iozone_linux.o
+	gcc -c $(OPT) -Dunix -DHAVE_ANSIC_C -DASYNC_IO -D_LARGEFILE64_SOURCE \
+		-D_FILE_OFFSET_BITS=64 -DSHARED_MEM -Dlinux libasync.c -o libasync.o 
+	gcc -c $(OPT) -Dunix -DHAVE_ANSIC_C -DASYNC_IO -D_LARGEFILE64_SOURCE \
+		-D_FILE_OFFSET_BITS=64 -DSHARED_MEM -Dlinux libbif.c -o libbif.o
 
 iozone_AIX.o: iozone.c libbif.c 
 	@echo ""
